### DISCUSSION TITLE
Miller indices of Specimen Z axis

### DISCUSSION THREAD

**Discussion by j-puerstl** [mtex-toolbox/mtex#1375] [Ask Anything]
Hello!

I have an EBSD orientation map and wanted to extract my specimen Z axis (i.e. the normal direction/loading axis) in the crystal coordinates of each grain. I was thinking of extracting this from an IPF map or by doing something along the lines of

r = vector3d(0,0,1)

r_crystal = inv(grains.meanOrientation) * r

but I'm not quite sure where the actual coordinates would be displayed.

Thank you! :)

**Comment by kilir** [comment]
Hi,
the direction in crystal coordinates, are of type Miller. You could for example plot them `plot(r_crystal)` in spherical projection.
Cheers,
Rüdiger

**Reply by j-puerstl** [reply]
Hi Rüdiger, 

Thanks a lot for your fast response!

Just to double check - I need to get the numerical values for the Miller Indices of Z as Input for another software: Do the x y and z values stored in r_crystal correspond to the Miller Indices? For a hexagonal material, would they be the Miller Indices corresponding to the hexagonal unit cell (defined by CS) or the orthonormal cell? What would uvw and hkl stored in r_crystal correspond to? 

Thanks a lot! :)

**Reply by kilir** [reply]
Hi,
I'm not entirely clear what you mean by numerical values for Millers, however `r_crystal.w`, respectively `r_crystal.l` will give you the 'w' respectively 'l' of the Miller, 'r_crystal.z' will give you 'z' as in Cartesian coordinates obtained from the normalized Miller.

I hope that helps.

Cheers,
Rüdiger
